### ISSUE TITLE
Implementation

### ISSUE THREAD

**Issue by sara-farahani** [thudzj/NeuralEigenfunctionSegmentor#1]
Hi! I have a problem with this implementation. In line 71 of the engine.py file, you have computed cross-entropy loss between the predicted segmentation map and the ground truth (loss = criterion (seg_pred, seg_gt)), and added it to the total loss (loss_value = loss.item() + neuralef_loss.item() + neuralef_reg.item()). My problem is that based on your paper, your method is unsupervised. Why have you used cross entropy? I tried to train the model without this loss, and the results were acceptable, but the accuracy decreased as expected. Is it like an option for having better results in implementation? I would appreciate it if you could help me with what is wrong.
